### PR TITLE
Add "selected_company" to "all_potential_upgrades"

### PR DESCRIPTION
**Pull request by carlorusso**

### CODE REVIEW

**Comment by tobymao** [comment]
why are some of these selected_company: and some selected_company = ? make it consistent

**Comment by carlorusso** [comment]
> why are some of these selected_company: and some selected_company = ? make it consistent

done

**Comment by carlorusso** [comment]
@tobymao is ok to merge this?

File: .rubocop.yml [hunk 1]
```diff
@@ -78,6 +78,10 @@ Lint/UnexpectedBlockArity:
 Lint/UnmodifiedReduceAccumulator:
   Enabled: true
 
+Lint/UnusedMethodArgument:
```
Comment (tobymao) [inline .rubocop.yml:81]: why did you change this?

Comment (carlorusso) [reply .rubocop.yml:81]: to simplify this:
      def all_potential_upgrades(tile, tile_manifest: false) # rubocop:disable Lint/UnusedMethodArgument
to this:
       def all_potential_upgrades(tile, tile_manifest: false, selected_company: nil)

we have 4 not used reference inside the code like this, and the keyword parameter cannot be rewritten with a underscore prefix so rubicon complains about that.
If you think is better to avoid to have keywords parameters not used I can change all of them to normal parameters with default

File: lib/engine/game/g_1873/game.rb [hunk 2]
```diff
@@ -1388,7 +1388,6 @@ def sellable_bundles(player, corporation)
           bundles.select { |bundle| @round.active_step.can_sell?(player, bundle) }
         end
 
-        # rubocop:disable Lint/UnusedMethodArgument
```
Comment (roseundy) [inline lib/engine/game/g_1873/game.rb:1391]: Why was this removed?

Comment (tobymao) [reply lib/engine/game/g_1873/game.rb:1391]: he removed the lint check
